### PR TITLE
Connect to MongoDB using mongoose

### LINKED ISSUE #7: Connect to DB
# User Story

As a user, I want to create a personal account so that all my data would by saved safe and independent from the app.

# To Do

* [x] Install mongoose
* [x] Create .env file with environment variables
* [x] Connect to Mongo DB using auth key from .env and mongoose
* [x] Create a User Model

### PR DESCRIPTION
**Pull request by berlinum**
Create a draft User model and route for user-authentication

Closes #7

### CODE REVIEW

**Comment by Johannafendrich** [comment]
looks like there is an issue with your error handling

File: lib/routes/auth.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const { Router } = require('express');
+
+const router = Router();
+
+// /api/auth/register
+router.post('/register', async (request, response) => {
+  try {
+    //const { email, password } = request.body;
+  } catch (error) {
```
Comment (marwinburesch) [inline lib/routes/auth.js:9]: `error` is an unused variable, that's why your deepscan check failed as well :)

File: .env [hunk 2]
```diff
@@ -0,0 +1,2 @@
+PORT=8080
+DB_URL=mongodb+srv://berlinum:XpYXHlabIFVtD3Qq@bricks-vv3is.mongodb.net/test?retryWrites=true&w=majority
```
Comment (lmachens) [inline .env:2]: This file should not be in git

Comment (berlinum) [reply .env:2]: Absolut! 🙈

File: index.js [hunk 3]
```diff
@@ -1,13 +1,37 @@
...
+
 const express = require('express');
 const path = require('path');
+const mongoose = require('mongoose');
+const auth = require('./lib/routes/auth');
 
 const port = process.env.PORT || 8080;
 const app = express();
 
+app.use('api/auth', auth);
```
Comment (lmachens) [inline index.js:12]: be careful here. 
the `/` is missing and would be an issue if you move files around.

```suggestion
app.use('/api/auth', auth);
```

**Pull request merged**